Edit HRMP opening info (#5494)
w3f/polkadot-wiki: docs/build/build-hrmp-channels.md

slug: ../build-hrmp-channels
---

[HRMP](../learn/learn-xcm-transport.md#hrmp-xcmp-lite) has the same interface and functionality as
[XCMP](../learn/learn-xcm-transport.md#xcmp-cross-chain-message-passing) but is much more demanding
on resources since it passes all messages via the Relay Chain. When XCMP is implemented on Polkadot,
HRMP is planned to be deprecated and phased out in favor of it.

## Opening HRMP channel: parachain to parachain

> To open a channel from one parachain to another, that is not system chains on Polkadot, the

Reviewer: ```suggestion
To open a channel from one parachain to another, that is not a system chain on Polkadot, the
```

> governance of each parachain needs to declare its intent to open a channel on the Relay Chain and

Reviewer: ```suggestion
governance of each parachain needs to declare its intent to open a channel on the Relay Chain through a `Transact` XCM message with parachain origin, and
```

Reviewer: From my understanding, this is done through a transact XCM, so it should be clearly communicated

the second chain needs to accept and do the same.

> To do this, the calls to be dispatched from both chains are:

Reviewer: ```suggestion
To do this, the calls to be included in the `Transact` message from both chains are:
```

Reviewer: Then we could go into more detail about the transact

1. `hrmp > hrmpInitOpenChannel(recipient, proposedMaxCapacity, proposedMaxMessageSize)`: this call
   initiates opening a channel from a parachain to a given recipient.

2. `hrmp > hrmpAcceptOpenChannel(sender)`: this call is for accepting the channel open request from
   the given sender.

Each parachain is to use its governance process to execute this. The call will be executed on the

> Relay Chain.

Reviewer: If it is to be made only through governance, then why do we have the xcm instruction [HrmpNewChannelOpenRequest](https://paritytech.github.io/xcm-docs/journey/channels-and-bridges.html#hrmpnewchannelopenrequest) sent by the relay as a notification?
At Polimec we are using this to do automatic channel opening, without going through governance.

Reviewer: Although we had to modify the xcm-executor because it had that instruction left empty, so maybe that change can happen once the executor adds an implementation for those instructions 🤔

## Opening HRMP Channels with System Parachains

Opening an HRMP channel with a system parachain requires an
[OpenGov referendum](../learn/learn-guides-polkadot-opengov.md) using the
[General Admin Track](../learn/learn-polkadot-opengov-origins.md#general-admin). Like all other
governance proposals, proposers should follow best practices like opening a discussion on
[Polkassembly](https://polkadot.polkassembly.io/) or [Subsquare](https://polkadot.subsquare.io/) and
then submitting the proposal on-chain.

Before submitting the proposal, both the system parachain's and the parachain's sovereign accounts
(both [system accounts](../learn/learn-account-advanced.md#system-accounts)) must have the channel
deposit of 20 DOT plus the
[existential deposit](../learn/learn-accounts.md#existential-deposit-and-reaping) and some extra

> funds to cover for fees. Anyone can send those funds via `balances.transferKeepAlive` call to the

Reviewer 2: This is not true. And what fees? There are no fees because there is no transaction.

Author: removed

> system parachain's sovereign account (if funds are not already available). System parachains have ID

Reviewer 2: Why the complexity? This can simply say, "at the time of execution, each parachain's sovereign account must have a free balance greater than or equal to the sum of the sender deposit, recipient deposit, and existential deposit."

Author: implemented

3.  Technical details of the proposal, including proposal parameters and technical details of this
    call (On Kusama, most proposals were designed as a batchAll calls):

>     - A `forceTransfer` of 20 DOT from the Treasury to AssetHub sovereign account.

Reviewer 2: Well now the first sentence below is not true...

Reviewer 2: Actually not just the first sentence, basically everything is wrong.
